widgets/deployments/src/DeploymentsList.jsx

```diff
@@ -66,73 +58,24 @@ export default class DeploymentsList extends React.Component {
         }
     };
 
-    deleteDeployment = () => {
-        const { deployment } = this.state;
-        const { toolbox } = this.props;
-        this.hideModal();
-
-        if (!deployment) {
-            this.setError('Something went wrong, no deployment was selected for delete');
-            return;
-        }
-
-        toolbox.loading(true);
-
-        const actions = new Stage.Common.DeploymentActions(toolbox);
-        actions
-            .doDelete(deployment)
-            .then(() => {
-                this.setError(null);
-                toolbox.getEventBus().trigger('deployments:refresh');
-                toolbox.loading(false);
-            })
-            .catch(err => {
-                this.setError(err.message);
-                toolbox.loading(false);
-            });
+    actOnExecution = (execution, action, error) => {
+        this.setError(error);
     };
 
-    forceDeleteDeployment = () => {
-        const { deployment } = this.state;
-        const { toolbox } = this.props;
-        this.hideModal();
-
-        if (!deployment) {
-            this.setError('Something went wrong, no deployment was selected for delete');
-            return;
-        }
-
-        toolbox.loading(true);
-
-        const actions = new Stage.Common.DeploymentActions(toolbox);
-        actions
-            .doForceDelete(deployment)
-            .then(() => {
-                this.setError(null);
-                toolbox.getEventBus().trigger('deployments:refresh');
-                toolbox.loading(false);
-            })
-            .catch(err => {
-                this.setError(err.message);
-                toolbox.loading(false);
-            });
+    openExecuteModal = (deploymentId, workflowName) => {
+        this.setState({ deploymentId, workflowName }, () => this.setState({ showExecuteModal: true }));
     };
 
-    actOnExecution = (execution, action, error) => {
-        this.setError(error);
+    hideExecuteModal = () => {
+        this.setState({ showExecuteModal: false });
     };
 
-    showModal = (value, deployment, workflow) => {
-        this.setState({
-            deployment,
-            workflow: workflow || {},
-            modalType: workflow ? MenuAction.WORKFLOW_ACTION : value,
-            showModal: true
-        });
+    openActionModal = (deploymentId, actionName) => {
+        this.setState({ deploymentId }, () => this.setState({ activeAction: actionName }));
```

Comment:
Is there any reason why it couldn't be:
```suggestion
        this.setState({ deploymentId, activeAction: actionName });
```
?

Reply:
Initially I thought it would be better to change it in two steps, but actually it works fine (no errors in browser console and in the UI), so I simplified it.